### PR TITLE
Chore: Fix all Typescript strict null errors 

### PR DESCRIPTION
**Pull request by torkelo**

### CODE REVIEW

File: public/app/core/components/Select/MetricSelect.tsx [hunk 1]
```diff
@@ -7,7 +7,7 @@ import { Variable } from 'app/types/templates';
 const { Select } = LegacyForms;
 
 export interface Props {
-  onChange: (value: string) => void;
+  onChange: (value: string | undefined) => void;
```
Comment (kaydelaney) [inline public/app/core/components/Select/MetricSelect.tsx:10]: Could this just be
```suggestion
  onChange: (value?: string) => void;
```

Comment (torkelo) [reply public/app/core/components/Select/MetricSelect.tsx:10]: no, it's not the same. The argument is not optional, but can be undefined

Comment (kaydelaney) [reply public/app/core/components/Select/MetricSelect.tsx:10]: Cool, makes sense

**Review by kaydelaney — APPROVED** [review]

File: tsconfig.json [hunk 2]
```diff
@@ -10,7 +10,8 @@
     },
     "rootDirs": ["public/"],
     "typeRoots": ["node_modules/@types", "public/app/types"],
-    "allowJs": true
+    "allowJs": true,
+    "strictNullChecks": true
```
Comment (dprokop) [inline tsconfig.json:14]: <3

**Review by dprokop — APPROVED** [review]
Plenty to digest but haven't found anything suspicious. Great work @torkelo!

**Pull request merged**